processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java

```diff
@@ -597,6 +600,48 @@ public Type asRawType() {
         }
     }
 
+    public Accessor getReadAccessor(String propertyName) {
+        if ( isMapType() ) {
+            List<Type> typeParameters = getTypeParameters();
+            if ( typeParameters.size() == 2 && typeParameters.get( 0 ).isString() ) {
```

Comment:
perhaps move this check to a method `hasMapSignature`...

Reply:
Good point, I have extracted a method `hasStringMapSignature` for this.